channeldb/invoices.go

```diff
@@ -505,6 +505,61 @@ func (d *DB) LookupInvoice(paymentHash [32]byte) (Invoice, error) {
 	return invoice, nil
 }
 
+// FetchAllInvoicesWithPaymentHash returns all invoices and their payment hashes
+// currently stored within the database. If the pendingOnly param is true, then
+// only unsettled invoices and their payment hashes will be returned, skipping
+// all invoices that are fully settled.
+func (d *DB) FetchAllInvoicesWithPaymentHash(pendingOnly bool) (
+	[]Invoice, []lntypes.Hash, error) {
```

Comment:
Add `PaymentHash` to `Invoice`? (in-memory field only, not serialized to db) and unify with existing FetchAll method?

Reply:
Done, though imho, this may be problematic. Changes semantics as not all ChannelDB functions can full the PaymentRequest and some users may find this confusing.

Reply:
So after taking a turn to that solution, the current cleaned up commit structure only has `FetchAllInvoicesWithPaymentHashes` and returns an array of structs instread of a tuple of arrays as discussed offline.